packages/nimble-components/src/rich-text-viewer/index.ts

```diff
@@ -11,7 +17,89 @@ declare global {
 /**
  * A nimble styled rich text viewer
  */
-export class RichTextViewer extends FoundationElement {}
+export class RichTextViewer extends FoundationElement {
+    /**
+     * @public
+     * @remarks
+     * Accessor: uses the parsing logic to convert the input markdown string to render in the component.
+     */
+    public set markdown(value: string) {
+        this._markdown = value;
+        this.serializedContent = this.parseMarkdownToDOM(value);
+        this.updateView();
+    }
+
+    /**
+     * @public
+     * @remarks
+     * Accessor: gets the raw markdown string.
+     */
+    public get markdown(): string {
+        return this._markdown;
+    }
+
+    private _markdown = '';
+    private serializedContent?: HTMLElement | DocumentFragment;
+    private readonly markdownParser: MarkdownParser;
+    private readonly domSerializer: DOMSerializer;
+
+    public constructor() {
+        super();
+        this.domSerializer = DOMSerializer.fromSchema(schema);
+        this.markdownParser = this.initializeMarkdownParser();
+    }
+
+    /**
+     * @internal
+     */
+    public override connectedCallback(): void {
+        super.connectedCallback();
+        this.updateView();
+    }
+
+    private initializeMarkdownParser(): MarkdownParser {
```

Comment:
Thanks for all the detailed comments on this implementation! It will really help with maintainability in the future, especially if someone who is less familiar with markdown-it / prose mirror needs to work on this code.